suricata/update/commands/enablesource.py

```diff
@@ -71,25 +72,35 @@ def enable_source():
         key, val = param.split("=", 1)
         opts[key] = val
 
-    source = source_index.get_sources()[name]
-
-    if "subscribe-url" in source:
-        print("The source %s requires a subscription. Subscribe here:" % (name))
-        print("  %s" % source["subscribe-url"])
-
     params = {}
-    if "parameters" in source:
-        for param in source["parameters"]:
-            if param in opts:
-                params[param] = opts[param]
-            else:
-                prompt = source["parameters"][param]["prompt"]
-                while True:
-                    r = input("%s (%s): " % (prompt, param))
-                    r = r.strip()
-                    if r:
-                        break
-                params[param] = r.strip()
+    if updating:
+        source = yaml.safe_load(open(sources.get_enabled_source_filename(name), "rb"))
+        if "params" in source:
+            params = source["params"]
+            for old_param in source["params"]:
+                if old_param in opts and source["params"][old_param] != opts[old_param]:
+                    logger.info("Updating source parameter '%s': '%s' -> '%s'." % (
+                        old_param, source["params"][old_param], opts[old_param]))
+                    params[old_param] = opts[old_param]
+    else:
+        source = source_index.get_sources()[name]
+
+        if "subscribe-url" in source:
+            print("The source %s requires a subscription. Subscribe here:" % (name))
+            print("  %s" % source["subscribe-url"])
+
+        if "parameters" in source:
+            for param in source["parameters"]:
+                if param in opts:
+                    params[param] = opts[param]
+                else:
+                    prompt = source["parameters"][param]["prompt"]
+                    while True:
+                        r = input("%s (%s): " % (prompt, param))
+                        r = r.strip()
+                        if r:
+                            break
+                    params[param] = r.strip()
```

Comment:
Since not all of the code in the above added block is newly added or modified, we could probably do it in a cleaner way following _Flat is better than nested._ from the [Zen of Python](https://www.python.org/dev/peps/pep-0020). ;)

So, something like, leave the code that is already there and do

```
+    if updating:
+        source = yaml.safe_load(open(sources.get_enabled_source_filename(name), "rb"))
+    else:
+        source = source_index.get_sources()[name]
+
+    if "params" in source:
+        params = source["params"]
+        for old_param in source["params"]:
+            if old_param in opts and source["params"][old_param] != opts[old_param]:
+                logger.info("Updating source parameter '%s': '%s' -> '%s'." % (
+                    old_param, source["params"][old_param], opts[old_param]))
+                params[old_param] = opts[old_param]
```

Will it make sense or mess up with your logic?

Reply:
I think this will work. I have also tested the changed code a bit in various situations. I'll address this and send a new PR. Thanks for taking a look!